elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/AggregationDataStoreTransactionTest.java

```diff
@@ -0,0 +1,151 @@
+import com.yahoo.elide.datastores.aggregation.example.PlayerStats;
+import com.yahoo.elide.datastores.aggregation.framework.SQLUnitTest;
+import com.yahoo.elide.datastores.aggregation.query.Cache;
+import com.yahoo.elide.datastores.aggregation.query.Query;
+import com.yahoo.elide.datastores.aggregation.query.QueryKeyExtractor;
+import com.yahoo.elide.datastores.aggregation.query.QueryResult;
+import com.yahoo.elide.request.EntityProjection;
+import com.yahoo.elide.request.Pagination;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Collections;
+
+@ExtendWith(MockitoExtension.class)
+class AggregationDataStoreTransactionTest extends SQLUnitTest {
+
+    @Mock private QueryEngine queryEngine;
+    @Mock private QueryEngine.Transaction qeTransaction;
+    @Mock private RequestScope scope;
+    @Mock private Cache cache;
+
+    private Query query = Query.builder().table(playerStatsTable).build();
+    private final String queryKey = QueryKeyExtractor.extractKey(query);
+    private static final Iterable<Object> DATA = Collections.singletonList("xyzzy");
+
+    // inject our own query instead of using buildQuery impl
+    private class MyAggregationDataStoreTransaction extends AggregationDataStoreTransaction {
+
+        public MyAggregationDataStoreTransaction(QueryEngine queryEngine, Cache cache) {
+            super(queryEngine, cache);
+        }
+
+        @Override
+        protected Query buildQuery(EntityProjection entityProjection, RequestScope scope) {
+            return query;
+        }
+    }
+
+    @BeforeAll
+    public static void beforeAllTests() {
+        SQLUnitTest.init();
+    }
+
+    @BeforeEach
+    public void setUp() {
+        when(queryEngine.beginTransaction()).thenReturn(qeTransaction);
+    }
+
+    @Test
+    void loadObjectsPopulatesCache() {
+        QueryResult queryResult = QueryResult.builder().data(DATA).build();
+        when(queryEngine.getTableVersion(playerStatsTable, qeTransaction)).thenReturn("foo");
+        when(queryEngine.executeQuery(query, qeTransaction)).thenReturn(queryResult);
+        AggregationDataStoreTransaction transaction = new MyAggregationDataStoreTransaction(queryEngine, cache);
+        EntityProjection entityProjection = EntityProjection.builder().type(PlayerStats.class).build();
+
+        assertEquals(DATA, transaction.loadObjects(entityProjection, scope));
+
+        String cacheKey = "foo;" + queryKey;
+        Mockito.verify(cache).get(cacheKey);
+        Mockito.verify(cache).put(cacheKey, queryResult);
+        Mockito.verifyNoMoreInteractions(cache);
+    }
+
+    @Test
+    void loadObjectsUsesCache() {
+        QueryResult queryResult = QueryResult.builder().data(DATA).build();
```

Comment:
Does this test execution depend on the prior test executing first?  I think we should make each test stand on its own.

Reply:
No, the fields annotated with `@Mock`, such as the cache, get reset for each test, and the rest of the fields used in the tests are immutable. So `loadObjectsUsesCache` is inserting a cache entry before this test with `when(cache.get(anyString())).thenReturn(queryResult);`.